internal/cmd/skupper/link/link.go

```diff
@@ -15,15 +19,108 @@ func NewCmdLink() *cobra.Command {
 skupper link status`,
 	}
 
-	linkExportCommand := kube.NewCmdLinkGenerate()
-	linkUpdateCommand := kube.NewCmdLinkUpdate()
-	linkStatusCommand := kube.NewCmdLinkStatus()
-	linkDeleteCommand := kube.NewCmdLinkDelete()
+	cmd.AddCommand(CmdLinkGenerateFactory(config.GetPlatform()))
+	cmd.AddCommand(CmdLinkUpdateFactory(config.GetPlatform()))
+	cmd.AddCommand(CmdLinkStatusFactory(config.GetPlatform()))
+	cmd.AddCommand(CmdLinkDeleteFactory(config.GetPlatform()))
 
-	cmd.AddCommand(&linkExportCommand.CobraCmd)
-	cmd.AddCommand(&linkUpdateCommand.CobraCmd)
-	cmd.AddCommand(&linkDeleteCommand.CobraCmd)
-	cmd.AddCommand(&linkStatusCommand.CobraCmd)
+	return cmd
+}
+
+func CmdLinkGenerateFactory(configuredPlatform types.Platform) *cobra.Command {
+	kubeCommand := kube.NewCmdLinkGenerate()
+	nonKubeCommand := non_kube.NewCmdLinkGenerate()
+
+	cmdLinkGenerateDesc := common.SkupperCmdDescription{
+		Use:   "generate",
+		Short: "Generate a new link resource in a yaml file",
+		Long: `Generate a new link resource with the data needed from the target site. The resultant
+output needs to be applied in the site in which we want to create the link.`,
+	}
+
+	cmd := common.ConfigureCobraCommand(configuredPlatform, cmdLinkGenerateDesc, kubeCommand, nonKubeCommand)
+	cmdFlags := common.CommandLinkGenerateFlags{}
+	cmd.Flags().StringVar(&cmdFlags.TlsSecret, common.FlagNameTlsSecret, "", common.FlagDescTlsSecret)
+	cmd.Flags().StringVar(&cmdFlags.Cost, common.FlagNameCost, "1", common.FlagDescCost)
+	cmd.Flags().StringVarP(&cmdFlags.Output, common.FlagNameOutput, "o", "yaml", common.FlagDescOutput)
```

Comment:
command has a (default "yaml")  
--output yaml    
--output json
--output
  is this the command format if using default ?  I guess that if you just specify --output it will default to yaml

Reply:
if you use only "--output" without specifying a value, the expected output is this one:

```
$skupper link generate  --tls-secret test2   -n south  --output                                             
Error: flag needs an argument: --output

flag needs an argument: --output
```
If you want to use the flag, you need to provide a valid value.
The default value is used when the user has not used the flag.

Reply:
ok my confusion as I was thinking output was being used like for the other commands.